### PR TITLE
Fix viewschedule bugs

### PR DESCRIPTION
**Pull request by vishandi**

### CODE REVIEW

File: src/main/java/seedu/address/model/schedule/Event.java [hunk 1]
```diff
@@ -227,6 +249,16 @@ public boolean willDateCollide(LocalDate date) {
         return false;
     }
 
+    /**
+     * Returns true if the event is happening sometime in {@param date}.
+     * @param date
```
Comment (tyanhan) [inline src/main/java/seedu/address/model/schedule/Event.java:254]: Line 254 can remove?

File: src/main/java/seedu/address/model/schedule/Event.java [hunk 2]
```diff
@@ -255,24 +287,44 @@ public boolean willDateTimeCollideEvent(LocalDate date, LocalTime time) {
      * @param date used to check.
      * @return an Event for that particular date.
```
Comment (tyanhan) [inline src/main/java/seedu/address/model/schedule/Event.java:288]: Need to update JavaDocs for this

**Review by tyanhan — CHANGES_REQUESTED** [review]

File: src/main/java/seedu/address/model/schedule/Event.java [hunk 3]
```diff
@@ -204,6 +201,33 @@ public boolean isValidDurationWithRecurFrequency() {
...
+        case BIWEEKLY:
+            return date.plusDays(14);
+        default:
+            logger.warning(String.format(MISSING_RECUR_FREQUENCY_CASE, recurFrequency));
+        }
+        return date;
+    }
+
+    /**
+     * Returns the next Event of the next time this event will occur, if it is a recurring event.
```
Comment (tyanhan) [inline src/main/java/seedu/address/model/schedule/Event.java:225]: Returns the next recurrence of this Event

**Comment by hanqinilnix** [comment]
Looks good to me

File: src/main/java/seedu/address/model/schedule/Event.java [hunk 4]
```diff
@@ -250,29 +279,45 @@ public boolean willDateTimeCollideEvent(LocalDate date, LocalTime time) {
...
-                        Duration.ofHours(24), recurFrequency);
-            } else if (nextEvent.getDate().isBefore(date) && nextEvent.getEndDate().isEqual(date)) {
-                Duration duration = Duration.between(LocalTime.MIDNIGHT, nextEvent.getEndTime());
-                return new Event(nextEvent.eventDescription, date, LocalTime.MIDNIGHT, duration, recurFrequency);
-            } else if (nextEvent.getDate().isEqual(date) && nextEvent.getEndDate().isAfter(date)) {
-                Duration duration = Duration.between(nextEvent.getTime(), LocalTime.MIDNIGHT).plusDays(1);
-                return new Event(nextEvent.eventDescription, date, nextEvent.getTime(), duration, recurFrequency);
-            } else {
-                return nextEvent;
+        if (date.isBefore(getDate()) || !willDateCollide(date)) {
```
Comment (hanqinilnix) [inline src/main/java/seedu/address/model/schedule/Event.java:291]: Perhaps you can add comments before the if statements to make the code more readable

**Comment by hanqinilnix** [comment]
Looks good

**Pull request merged**